### PR TITLE
Rename RoomBehaviorType2

### PR DESCRIPTION
**Pull request by mzxrules**
Contributions made in this pr are licensed under CC0

### CODE REVIEW

File: include/z64scene.h [hunk 1]
```diff
@@ -161,20 +161,21 @@ typedef enum RoomBehaviorType1 {
...
+typedef enum RoomPlayerFidgetType {
+    // Values mirror FidgetType
+    /* 0 */ ROOM_PLAYER_FIDGET_LOOK_AROUND,
+    /* 1 */ ROOM_PLAYER_FIDGET_COLD,
+    /* 2 */ ROOM_PLAYER_FIDGET_WARM,
+    /* 3 */ ROOM_PLAYER_FIDGET_HOT, // Enables hot room timer for the current room
+    /* 4 */ ROOM_PLAYER_FIDGET_STRETCH_1,
+    /* 5 */ ROOM_PLAYER_FIDGET_STRETCH_2,
+    /* 6 */ ROOM_PLAYER_FIDGET_STRETCH_3
+} RoomPlayerFidgetType;
```
Comment (fig02) [inline include/z64scene.h:173]: I dont really agree with naming these after player fidget types. Its moreso that links fidget type depends on a property of the room, not that it *is* the property. I dont have a good suggestion for what that should be at the moment though.

Comment (fig02) [reply include/z64scene.h:173]: I think the fact that `ROOM_PLAYER_FIDGET_HOT` enables the hot room timer tells this much. Its like a "room environment type" of some kind. But Ive not taken the time to compare this to roombehavior1 and see if its a good name.

Comment (mzxrules) [reply include/z64scene.h:173]: Looked into it, and roombehaviorType1 is unrelated. 

- `ROOM_BEHAVIOR_TYPE1_0` has PRINTF "camera: room type: default set field" associated with it. My scene data dumps show the value is used for outside locations and unused dungeon room layers
- `ROOM_BEHAVIOR_TYPE1_1` has no PRINTF but assigns `CAM_SET_DUNGEON0`. The value is set in dungeon rooms and dungeon-like rooms like Dampe's Grave (but not the windmill). It notably disables the use of Sun's Song regardless of restriction flags state.
- `ROOM_BEHAVIOR_TYPE1_2` is associated with default case PRINTF "camera: room type: default set etc". It's used in indoor rooms, capping Link's run speed and preventing him from jumping or attacking
- `ROOM_BEHAVIOR_TYPE1_5` is associated with default case PRINTF "camera: room type: default set etc". It's used exclusively boss rooms, and blocks `Environment_AdjustLights` and `MagicDark_DimLighting` from adjusting the lighting
- `ROOM_BEHAVIOR_TYPE1_3` is unused. It appears to disable color dithering when drawing the room and other things
- `ROOM_BEHAVIOR_TYPE1_4` is unused. It appears to block setting `CAM_SET_HORSE` when getting on a horse.

Comment (cadmic) [reply include/z64scene.h:173]: I agree naming it after environment seems better. It seems reasonable that link's fidget animations are based on his environment (I know mine are IRL). Maybe ROOM_PLAYER_FIDGET_LOOK_AROUND could be "normal" and ROOM_PLAYER_FIDGET_STRETCH_1/2/3 could be something like "safe" rooms (they're indoors and towns but I couldn't find a clear pattern, maybe you can though)

Comment (mzxrules) [reply include/z64scene.h:173]: I would like to do that, but to me there are just too many unknowns to reliably give proper names to half of the types. By naming them after their associated fidget animation, you at least aren't given the false impression that the property is doing something more than configure the player idle animation. 

As for `FIDGET_STRETCH_1/2/3`, the idea that they are set in "safe" rooms doesn't really seem to make sense. Kakariko and Kokiri both are relatively safe places, but neither use type 4. Additionally, when you consider MM and the fact the system goes largely unchanged the "safe" room idea breaks down completely. Southern Swamp (Poisoned and Restored), Great Bay Coast, Zora cape are some of the places that set type 4.

Comment (fig02) [reply include/z64scene.h:173]: I cannot get behind naming these as "room fidget types". I propose the following:
```
typedef enum RoomEnvironmentType {
    /* 0 */ ROOM_ENV_NEUTRAL,
    /* 1 */ ROOM_ENV_COLD,
    /* 2 */ ROOM_ENV_WARM,
    /* 3 */ ROOM_ENV_HOT,
    /* 4 */ ROOM_ENV_UNK_STRETCH_1, // Remaining types temporarily named after their corresponding fidget type
    /* 5 */ ROOM_ENV_UNK_STRETCH_2,
    /* 6 */ ROOM_ENV_UNK_STRETCH_3
} RoomEnvironmentType;
```
Someone may see the matrix some day and figure out the pattern with the stretch ones

Comment (mzxrules) [reply include/z64scene.h:173]: If you do that, aren't you better off just calling `ROOM_ENV_NEUTRAL`  `ROOM_ENV_LOOK_AROUND`? There isn't any special meaning to `ROOM_ENV_NEUTRAL`

Comment (fig02) [reply include/z64scene.h:173]: I think its easy enough to tell that its "neutral" even though there are no distinguishing features about it. The look around anim is the most "default"/"normal"/"neutral" looking of the environmental fidget types. Even the fact that its type 0 hints toward this.

It does suck that we dont have more information to go off of here, but we can try to name it after an environment type with the crumbs of info we have

**Review by cadmic — APPROVED** [review]

File: include/z64scene.h [hunk 2]
```diff
@@ -161,20 +161,21 @@ typedef enum RoomBehaviorType1 {
...
-    /* 0 */ ROOM_BEHAVIOR_TYPE2_0,
-    /* 1 */ ROOM_BEHAVIOR_TYPE2_1,
-    /* 2 */ ROOM_BEHAVIOR_TYPE2_2,
-    /* 3 */ ROOM_BEHAVIOR_TYPE2_3,
-    /* 4 */ ROOM_BEHAVIOR_TYPE2_4,
-    /* 5 */ ROOM_BEHAVIOR_TYPE2_5,
-    /* 6 */ ROOM_BEHAVIOR_TYPE2_6
-} RoomBehaviorType2;
+typedef enum RoomEnvironmentType {
+    // Values align with FidgetType
```
Comment (fig02) [inline include/z64scene.h:165]: I think its more appropriate to say that the first few fidget types align with the room environment types. We can put the types in comments in the fidget enum, for example:
```c
typedef enum FidgetType {
    /* 0x00 */ FIDGET_LOOK_AROUND, // ROOM_ENV_DEFAULT
    /* 0x01 */ FIDGET_COLD,        // ROOM_ENV_COLD
    /* 0x02 */ FIDGET_WARM,        // ROOM_ENV_WARM
    /* 0x03 */ FIDGET_HOT,         // ROOM_ENV_HOT (same animations as FIDGET_WARM)
    /* 0x04 */ FIDGET_STRETCH_1,   // ROOM_ENV_UNK_STRETCH_1
    /* 0x05 */ FIDGET_STRETCH_2,   // ROOM_ENV_UNK_STRETCH_2 (same animations as FIDGET_STRETCH_1)
    /* 0x06 */ FIDGET_STRETCH_3,   // ROOM_ENV_UNK_STRETCH_3 (same animations as FIDGET_STRETCH_1)
    /* 0x07 */ FIDGET_CRIT_HEALTH_START,
    /* 0x08 */ FIDGET_CRIT_HEALTH_LOOP,
    /* 0x09 */ FIDGET_SWORD_SWING,
    /* 0x0A */ FIDGET_ADJUST_TUNIC,
    /* 0x0B */ FIDGET_TAP_FEET,
    /* 0x0C */ FIDGET_ADJUST_SHIELD,
    /* 0x0D */ FIDGET_SWORD_SWING_TWO_HAND
} FidgetType;
```

**Review by fig02 — APPROVED** [review]

**Pull request merged**